cdap-watchdog/src/main/java/co/cask/cdap/metrics/guice/MetricsClientRuntimeModule.java

```diff
@@ -41,11 +38,14 @@ public Module getInMemoryModules() {
     return new PrivateModule() {
       @Override
       protected void configure() {
-        bind(MetricDatasetFactory.class).to(DefaultMetricDatasetFactory.class).in(Scopes.SINGLETON);
-        bind(MetricStore.class).to(DefaultMetricStore.class);
+        // Install the MetricsStoreModule as private bindings and expose the MetricStore.
+        // Both LocalMetricsCollectionService and the AppFabricService needs it
+        install(new MetricsStoreModule());
         expose(MetricStore.class);
```

Comment:
As the comment said, because "// Both LocalMetricsCollectionService and the AppFabricService needs it".